### PR TITLE
Infer proper types within tuples.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/14277

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 1]
```diff
@@ -335,6 +336,22 @@ private IEnumerable<TypeInferenceInfo> InferTypeInAnonymousObjectCreation(Anonym
                 return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
             }
 
+            private IEnumerable<TypeInferenceInfo> InferTypeInTupleExpression(
+                TupleExpressionSyntax tupleExpression, SyntaxToken previousToken)
+            {
+                if (previousToken == tupleExpression.OpenParenToken)
+                    return InferTypeInTupleExpression(tupleExpression, tupleExpression.Arguments[0]);
+
+                if (previousToken.Kind() == SyntaxKind.CommaToken)
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]:
```suggestion
                if (previousToken.IsKind(SyntaxKind.CommaToken))
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs [hunk 2]
```diff
@@ -335,6 +336,22 @@ private IEnumerable<TypeInferenceInfo> InferTypeInAnonymousObjectCreation(Anonym
...
+                {
+                    var argsAndCommas = tupleExpression.Arguments.GetWithSeparators();
+                    var commaIndex = argsAndCommas.IndexOf(previousToken);
+                    return InferTypeInTupleExpression(tupleExpression, (ArgumentSyntax)argsAndCommas[commaIndex + 1]);
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: ❔ Will this break for invalid syntax where the last argument is omitted?

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: Will add test.  :)

Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs]: test added.

**Pull request merged**